packages/react-dom/src/client/setInnerHTML.js

```diff
@@ -27,10 +28,20 @@ const setInnerHTML = createMicrosoftUnsafeLocalFunction(function(
   // IE does not have innerHTML for SVG nodes, so instead we inject the
   // new markup in a temp node and then move the child nodes across into
   // the target node
-
   if (node.namespaceURI === Namespaces.svg && !('innerHTML' in node)) {
     reusableSVGContainer =
       reusableSVGContainer || document.createElement('div');
+    if (__DEV__) {
+      warningWithoutStack(
+        typeof window.TrustedTypes === 'undefined',
+        "Using 'dangerouslySetInnerHTML' in an svg element with " +
+          'Trusted Types enabled in an Internet Explorer will cause ' +
+          'the trusted value to be converted to string. Assigning string ' +
+          "to 'innerHTML' will throw an error if Trusted Types are enforced. " +
+          "You can try to wrap your svg element inside a div and use 'dangerouslySetInnerHTML' " +
+          'on the enclosing div instead.',
```

Comment:
1. In which situations can IE possibly have TT support? Is this about new Edge? Doesn't it use Blink, meaning it no longer has this issue?

2. Regardless, we usually don't want to warn about a pattern that's broken in one browser, just in that browser. Since developers would usually develop with their browser of choice and only later test it. Generally we want to prevent developers from writing potentially broken code early — instead of waiting for them to test in IE. If React API lets us do something, ideally it would work in all browsers. Let's figure out (1) first though.

Reply:
> 1. In which situations can IE possibly have TT support? Is this about new Edge? Doesn't it use Blink, meaning it no longer has this issue?

IE applications can enable TT  if they load the [polyfill](https://github.com/WICG/trusted-types). 

> 2\. Regardless, we usually don't want to warn about a pattern that's broken in one browser, just in that browser. Since developers would usually develop with their browser of choice and only later test it.

Yeah, that is a good point. Easy way is to just log in every browser? The problem is that we couldn't figure out a solution without without introducing a policy for React that blesses hacky solution for IE.

Reply:
>IE applications can enable TT if they load the polyfill.

Does that polyfill enable enforcement?

Reply:
Whether Trusted Types are enforced or just in report only mode is up to the application. It can set the mode using the CSP header. The polyfill itself just provides Trusted Types API and patches the DOM sinks. _([reference](https://github.com/WICG/trusted-types#browsers))_

Reply:
Do I understand correctly that this issue would only affect websites enable TT polyfill in enforcement mode on IE? Since if it's not being enforced, `'<svg>' + html + '</svg>'` should just stringify?

Reply:
Just to clarify -- The problem is not the stringification. Expression `'<svg>' + html + '</svg>'`is stringified independendly of whether Trusted Types are enforced or not. The problem is when you try to assign the **string** value to DOM sink _(e.g. innerHTML)_.

Otherwise your understanding is correct. It will break **iff**:
1) application **enforces** Trusted Types
2) uses `dangerouslySetInnerHTML` on svg nodes
3) runs on IE